lib/experimental/media/load.php

```diff
@@ -0,0 +1,346 @@
+<?php
+/**
+ * Adds media-related experimental functionality.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Returns a list of all available image sizes.
+ *
+ * @return array Existing image sizes.
+ */
+function gutenberg_get_all_image_sizes(): array {
+	$sizes = wp_get_registered_image_subsizes();
+
+	foreach ( $sizes as $name => &$size ) {
+		$size['height'] = (int) $size['height'];
+		$size['width']  = (int) $size['width'];
+		$size['name']   = $name;
+	}
+	unset( $size );
+
+	return $sizes;
+}
+
+/**
+ * Returns the default output format mapping for the supported image formats.
+ *
+ * @return array<string,string> Map of input formats to output formats.
+ */
+function gutenberg_get_default_image_output_formats() {
+	$input_formats = array(
+		'image/jpeg',
+		'image/png',
+		'image/gif',
+		'image/webp',
+		'image/avif',
+		'image/heic',
+	);
+
+	$output_formats = array();
+
+	foreach ( $input_formats as $mime_type ) {
```

Comment:
Odd to see this called in a loop for all types, I guess this is to gather data for the client side so it can handle any upload type? Since only one type can be uploaded at once, I would expect this only need a single call.

Reply:
The  thing with the `image_editor_output_format` filter is that it passes a file name and a source mime type as 2nd & 3rd arg. Calling this in a loop here so that any plugin using this filter gets expected args. Otherwise this wouldn't be necessary.